synapse/storage/roommember.py

```diff
@@ -639,6 +661,39 @@ def f(txn):
         count = yield self.runInteraction("did_forget_membership", f)
         defer.returnValue(count == 0)
 
+    @cached()
+    def get_forgotten_rooms_for_user(self, user_id):
+        """Gets all rooms the user has forgotten.
+
+        Args:
+            user_id (str)
+
+        Returns:
+            Deferred[set[str]]
+        """
+
+        def _get_forgotten_rooms_for_user_txn(txn):
+            # This is a slightly convoluted query that first looks up all rooms
```

Comment:
so is this in aid of trying to support unforgetting of rooms? is that a thing that's possible?

Reply:
Yup, you can just rejoin the room (I believe)

Reply:
actually could you update the comment to make this a bit clearer?